apps/backend/src/frameworks/opencage.service.ts

```diff
@@ -0,0 +1,44 @@
+import * as https from 'https';
+
+export class OpenCageService {
+  async reverseGeocoding(latitude: string, longitude: string): Promise<any> {
+    if (!latitude || !longitude) {
+      throw new Error('Latitude and longitude must be provided.');
+    }
+
+    const url = `https://api.opencagedata.com/geocode/v1/json?q=${latitude}+${longitude}&key=${process.env.OPENCAGE_KEY}`;
+
+    return new Promise((resolve, reject) => {
```

Comment:
once you replace that with the built-in HTTP module, please use async/await instead of promises.